langchain4j-workers-ai/src/main/java/dev/langchain4j/model/workersai/WorkersAiImageModel.java

```diff
@@ -0,0 +1,188 @@
+        ensureNotBlank(prompt, "Prompt");
+        ensureNotNull(image, "Image");
+        ensureNotNull(mask, "Mask");
+        return new Response<>(convertAsImage(executeQuery(prompt, mask, image)), null, FinishReason.STOP);
+    }
+
+    /**
+     * Generate image and save to file.
+     *
+     * @param prompt
+     *      current prompt
+     * @param destinationFile
+     *      local file
+     * @return
+     *      response with the destination file
+     */
+    public Response<File> generate(String prompt, String destinationFile) {
+        ensureNotBlank(prompt, "Prompt");
+        ensureNotBlank(destinationFile, "Destination file");
+        try {
+            byte[] image = executeQuery(prompt, null, null);
+            try (FileOutputStream fileOutputStream = new FileOutputStream(destinationFile)) {
+                fileOutputStream.write(image);
+            }
+            return new Response<>(new File(destinationFile), null, FinishReason.STOP);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Execute query.
+     *
+     * @param prompt
+     *     prompt.
+     * @return
+     *    image.
+     */
+    private byte[] executeQuery(String prompt, Image image, Image mask) {
+        try {
+            // Mapping inbound
+            WorkersAiImageGenerationRequest imgReq = new WorkersAiImageGenerationRequest();
+            imgReq.setPrompt(prompt);
+            if (image != null) {
+                if (image.url() != null) {
+                    imgReq.setImage(getPixels(image.url().toURL()));
+                    System.out.println("Adding Image");
+                }
+            }
+            if (mask != null) {
+                if (mask.url() != null) {
+                    imgReq.setMask(getPixels(mask.url().toURL()));
+                    System.out.println("Adding mask");
+                }
+            }
+
+            retrofit2.Response<ResponseBody> response = workerAiClient
+                    .generateImage(imgReq, accountIdentifier, modelName)
+                    .execute();
+
+            System.out.println(response.code() + " " + response.message());
+            if (response.isSuccessful() && response.body() != null) {
+                InputStream inputStream = response.body().byteStream();
+                ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+                int nRead;
+                byte[] data = new byte[1024];
+                while ((nRead = inputStream.read(data, 0, data.length)) != -1) {
+                    buffer.write(data, 0, nRead);
+                }
+                buffer.flush();
+                return buffer.toByteArray();
+            }
+            throw new IllegalStateException("An error occured while generating image.");
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public int[] getPixels(URL imageUrl) throws Exception {
```

Comment:
Is there any other way to do this?

Reply:
this one no idea, it is not documented by cloudFlare